CRM/MembershipExtras/Job/OfflineAutoRenewal/PaymentPlan.php

```diff
@@ -880,17 +897,24 @@ protected function calculateRenewedMembershipsStartDate() {
       }
 
       $membershipEndDate = new DateTime($lineItem['memberhsip_end_date']);
-      if (!isset($latestDate)) {
-        $latestDate = $membershipEndDate;
+      $isLineAutoRenewable = $lineItem['auto_renew'] == '1' ? 'auto_renewable' : 'not_auto_renewable';
+
+      if (!isset($latestEndDates[$isLineAutoRenewable])) {
```

Comment:
what if you have two memberships in the current tab that we set to "not" renew, and added only one to the next period tab, I think it make sense to pick the latest end date in such case too, but you are not doing and date comparison for that, so I think it make more sense to change these lines to :

```
$membershipEndDate = new DateTime($lineItem['memberhsip_end_date']);

if (empty($lineItem['auto_renew']) && $latestEndDates['not_auto_renewable'] < $membershipEndDate) {
$latestEndDates['not_auto_renewable'] = $membershipEndDate;
}


if (!empty($lineItem['auto_renew']) && $latestEndDates['auto_renew'] < $membershipEndDate) {
$latestEndDates['auto_renew'] = $membershipEndDate;
}

```

Reply:
I think this is equivalent to this part:
```
      if (!isset($latestEndDates[$isLineAutoRenewable])) {
        ...
      elseif ($latestEndDates[$isLineAutoRenewable] < $membershipEndDate) {
        $latestEndDates[$isLineAutoRenewable] = $membershipEndDate;
      }
```

I mean, if `$lineItem['auto_renew']` is empty, then my code translates to:
```
      if (!isset($latestEndDates['not_auto_renewable'])) {
        ...
      elseif ($latestEndDates['not_auto_renewable'] < $membershipEndDate) {
        $latestEndDates['not_auto_renewable'] = $membershipEndDate;
      }
```

Which is equivalent to what you wrote for the empty autor_renew flag scenario, without the `empty($lineItem['auto_renew'])`, which I don't need, as I already know it is empty. It just factorizes both cases in the same code.

In the scenario you describe, with the two memberships for current period that are not auto-renew, plus a new membership on the next period, the system will still pick the latest end date. For example:

- Membership 1, not auto-renew, ends on 2020-06-30
- Membership 2, not auto-renew, ends on 2020-12-31
- Membership 3 for new period.

In this case, the list of active memberships would be Membership 1 and Membership 2. At the end of the first cycle, the array with the end dates will be:
```
[
  'not_auto_renewable' => '2020-06-30',
  'auto_renew' => null,
]
```

At the end of the second cycle:
```
[
  'not_auto_renewable' => '2020-12-31',
  'auto_renew' => null,
]
```

Finally, since the latest end date for auto renewable memberships is null, it uses 2020-12-31, which is correct.